[14.0][FIX] cetmix_tower_server: Security restrictions on the command wizard path field

Aldeigja opened the pull request:
Fixing an [issue](https://github.com/cetmix/cetmix-tower/pull/59#pullrequestreview-2364938160) reported in #59 
Before this PR:

Group_user members can't run command wizards on servers due to field "path" restricitions

After this PR:

Members of group_user can run command wizards on servers

Aldeigja committed:
[FIX] cetmix_tower_server:

Before this commit:

Group_user members can't run command wizards on servers due to field "path" restrictions

After this commit:

Members of group_user can run command wizards on servers

ivs-cetmix reviewed (changes requested):
Please check the suggestions. Probably they will work faster.
Regarding tests, we need to ensure that commands run using wizard will be executed with correct path based on access group. So we need to check the execution results (command logs) for that.

ivs-cetmix commented:
/ocabot rebase

CetmixGitDrone commented:
Congratulations, PR rebased to [14.0-dev](https://github.com/cetmix/cetmix-tower/commits/14.0-dev).

ivs-cetmix commented:
Hi @norlinhenrik would appreciate you doing a functional review of this PR.

ivs-cetmix reviewed (approved):
Code and functional review LGTM. 
Please update PR and commit messages with the changes context.
I would suggest using "Before/after" pattern for that.

CetmixGitDrone commented:
This PR has the `approved` label and has been created more than 5 days ago. It should therefore be ready to merge by a maintainer (or a PSC member if the concerned addon has no declared maintainer). 🤖

Aldeigja committed:
[IMP] cetmix_tower_server: update the demo data to simplify and speed up the testing process

- add "Demo" user as follower to the server "Demo Server #2"
- change access level to "User" in SSH and Python commands
- create new flight plan with access level "User"

GabbasovDinar commented on cetmix_tower_server/demo/demo_data.xml:
```diff
@@ -243,11 +251,12 @@
+            if {{ branch }}:
+            COMMAND_RESULT={"exit_code": 0, "message": "Branch is defined!"}
+            else:
+            COMMAND_RESULT={"exit_code": -1, "message": "Branch is not defined!"}
```
why are these changes necessary here? also your changes is not python format 
please do not change the code formatting unless it is a bug, you are making an unnecessary update to git history.

Aldeigja replied:
I think it's some formatting done by pre-commit, I'll fix it.

GabbasovDinar commented on cetmix_tower_server/wizards/cx_tower_command_execute_wizard.py:
```diff
@@ -153,7 +152,11 @@ def execute_command_on_server(self):
-                self.command_id, sudo=self.use_sudo, path=self.path, **custom_values
+                self.command_id,
+                sudo=self.use_sudo,
+                path=self.env.user.has_group("cetmix_tower_server.group_manager")
```
`path=self.env.user ......` you can take it out of the loop so that you don't have to calculate this data every time.

GabbasovDinar reviewed (approved):
LGTM

CetmixGitDrone commented:
This PR has the `approved` label and has been created more than 5 days ago. It should therefore be ready to merge by a maintainer (or a PSC member if the concerned addon has no declared maintainer). 🤖

ivs-cetmix commented:
/ocabot merge nobump

CetmixGitDrone commented:
Hey, thanks for contributing! Proceeding to merge this for you.
Prepared branch [14.0-dev-ocabot-merge-pr-94-by-ivs-cetmix-bump-nobump](https://github.com/cetmix/cetmix-tower/commits/14.0-dev-ocabot-merge-pr-94-by-ivs-cetmix-bump-nobump), awaiting test results.

CetmixGitDrone commented:
Congratulations, your PR was merged at 3a0bb2d394cd79cf6ddc1dcdaace279955c43591. Thanks a lot for contributing to cetmix. ❤️
